packages/shopping/src/models/user.model.ts

```diff
@@ -51,6 +52,9 @@ export class User extends Entity {
   @hasOne(() => UserCredentials)
   userCredentials: UserCredentials;
 
+  @hasMany(() => UserRefreshtoken)
```

Comment:
Why not hasOne? Can user have multiple?

Reply:
Currently the `refreshToken` is newly generated at login time. As you can login multiple times or from multiple browsers/devices this needs to be a `hasMany` relation.

Alternatively we could share the `refreshToken` accross browsers/devices, than a `hasOne` would fit. But this might decrease security!?

WDYT?

Reply:
For simplicity, sharing a single token across devices would have a single point of failure, any detected breach would log out all devices. Could be annoying to user.

Your solution is more flexible/complex, but might think about identifying the authorised device linked to the refresh token maybe?  Identifying devices, allows the token to be individually invalidated (i.e. Lost device can have access revoked, although any serious security will remote wipe devices). If one token is compromised, is there any practical difference in access compromised?

Both solutions allow:
- Local - individual devices to be logged out(delete local token)
- Server - all devices to be logged out(expire one token vs. expire all tokens)
- Server - update permissions (update one token vs. update all tokens)

Multiple refresh token adds:
- Server - log out single device(given identity is known, expire the individual token)

Do what you think fits the scenario best.  I'm just offering thoughts for consideration.